### PR TITLE
[WIP] #r nuget improvements

### PR DESCRIPTION
**Pull request by jonsequitur**
This PR addresses #354, #583, #582, and #584.

### CODE REVIEW

File: NotebookExamples/Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
```
Comment (colombod) [inline NotebookExamples/Untitled.ipynb]: careful with untitled ntoebook

File: WorkspaceServer/Kernel/CSharpKernel.cs [hunk 2]
```diff
@@ -42,14 +43,13 @@ public class CSharpKernel : KernelBase, IExtensibleKernel
...
 
-
         public CSharpKernel()
         {
             _cancellationSource = new CancellationTokenSource();
             Name = DefaultKernelName;
         }
 
-        public ScriptOptions ScriptOptions { get; private set; } =
+        public ScriptOptions ScriptOptions { get; } =
```
Comment (colombod) [inline WorkspaceServer/Kernel/CSharpKernel.cs]: Do we need this property as public?

Comment (jonsequitur) [reply WorkspaceServer/Kernel/CSharpKernel.cs]: Probably not.

**Comment by eerhardt** [comment]
Should also fix https://github.com/dotnet/try/issues/458.

**Comment by jonsequitur** [comment]
Closing because #600 is better.

**Pull request closed without merging**